alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultAuthenticationTypeAccessor.java

```
@@ -0,0 +1,57 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.database.api;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.synopsys.integration.alert.common.enumeration.AuthenticationType;
+import com.synopsys.integration.alert.common.persistence.accessor.AuthenticationTypeAccessor;
+import com.synopsys.integration.alert.common.persistence.model.AuthenticationTypeDetails;
+import com.synopsys.integration.alert.database.user.AuthenticationTypeEntity;
+import com.synopsys.integration.alert.database.user.AuthenticationTypeRepository;
+
+@Component
+@Transactional
+public class DefaultAuthenticationTypeAccessor implements AuthenticationTypeAccessor {
+    private final AuthenticationTypeRepository authenticationTypeRepository;
+
+    @Autowired
+    public DefaultAuthenticationTypeAccessor(AuthenticationTypeRepository authenticationTypeRepository) {
+        this.authenticationTypeRepository = authenticationTypeRepository;
+    }
+
+    @Override
+    public Optional<AuthenticationTypeDetails> getAuthenticationTypeDetails(AuthenticationType authenticationType) {
+        Optional<AuthenticationTypeEntity> authenticationTypeEntity = authenticationTypeRepository.findById(authenticationType.getId());
+        return authenticationTypeEntity.map(entity -> new AuthenticationTypeDetails(entity.getId(), entity.getName()));
+    }
+
+    @Override
+    public AuthenticationType getAuthenticationType(Long id) {
+        return AuthenticationType.getById(id);
```

Should we allow this to throw unchecked exceptions if the id is invalid?

---

Actually, it looks like this reads from a map, so no exception, but it can return null. We might want to wrap it in an optional.